### PR TITLE
DialogFlow fulfillment support

### PR DESCRIPTION
**Pull request by jcanizalez**
Hi team,

Fixes hashicorp/terraform-provider-google#6086

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_dialogflow_fulfillment`
```

### CODE REVIEW

**Comment by jcanizalez** [comment]
@googlebot I signed it!

File: mmv1/products/dialogflow/api.yaml [hunk 1]
```diff
@@ -318,4 +318,74 @@ objects:
...
+              * TYPE_UNSPECIFIED: Feature type not specified.
+              * SMALLTALK: Fulfillment is enabled for SmallTalk.
+            values:
+              - :TYPE_UNSPECIFIED
+              - :SMALLTALK
+      - !ruby/object:Api::Type::NestedObject
+        name: 'genericWebService'
+        description: |
+          Represents configuration for a generic web service. Dialogflow supports two mechanisms for authentications: - Basic authentication with username and password. - Authentication with additional authentication headers.
+        properties:
```
Comment (rileykarson) [inline mmv1/products/dialogflow/api.yaml:375]: I think all these subfields can be made required?

File: mmv1/products/dialogflow/api.yaml [hunk 2]
```diff
@@ -318,4 +318,74 @@ objects:
...
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'type'
+            description: |
+              The type of the feature that enabled for fulfillment.
+              * TYPE_UNSPECIFIED: Feature type not specified.
+              * SMALLTALK: Fulfillment is enabled for SmallTalk.
+            values:
+              - :TYPE_UNSPECIFIED
```
Comment (rileykarson) [inline mmv1/products/dialogflow/api.yaml:369]:
```suggestion
```

`X_UNSPECIFIED` are an artifact of the docs generation, generally, and aren't accepted by the server. We can safely omit it, I think, unless you've succeeded in provisioning a resource w/ it set.

File: mmv1/products/dialogflow/api.yaml [hunk 3]
```diff
@@ -318,4 +318,74 @@ objects:
...
+        description: |
+          Whether fulfillment is enabled.
+      - !ruby/object:Api::Type::Array
+        name: 'features'
+        description: |
+         The field defines whether the fulfillment is enabled for certain features.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'type'
```
Comment (rileykarson) [inline mmv1/products/dialogflow/api.yaml:363]: This can be made required.

File: mmv1/products/dialogflow/api.yaml (hunk 1 above)
Comment (jcanizalez) [reply mmv1/products/dialogflow/api.yaml:375]: Hi @rileykarson checking documentation and only uri field is required so I am marking it as required https://cloud.google.com/dialogflow/es/docs/reference/rest/v2/projects.agent.environments#Environment.GenericWebService

File: mmv1/products/dialogflow/api.yaml (hunk 2 above)
Comment (jcanizalez) [reply mmv1/products/dialogflow/api.yaml:369]: Got it I will remove it

File: mmv1/products/dialogflow/api.yaml (hunk 3 above)
Comment (jcanizalez) [reply mmv1/products/dialogflow/api.yaml:363]: sure

**Comment by jcanizalez** [comment]
hi @rileykarson 

Thanks for your review I applied your suggestions

**Review by rileykarson — APPROVED** [review]
LGTM, thanks!

**Pull request merged**